src/fairseq2/datasets/parallel_text.py

```diff
@@ -33,7 +33,7 @@
 from fairseq2.typing import Device, override
 
 
-@dataclass(frozen=True)
+@dataclass(unsafe_hash=True)  # Due to FSDP, we cannot freeze.
```

Comment:
This is pretty interesting. Do you have a reference for this ?

Reply:
I don't think there is a written reference. In fact, this is mostly an unintentional side-effect of FSDP implementation, since it blindly tries to set attributes on model input/outputs.

Reply:
Thank you for the context!